app/helpers/concerns/foreman_remote_execution/job_templates_extensions.rb

```diff
@@ -4,7 +4,7 @@ def permitted_actions(template)
       original = super(template)
 
       if template.is_a?(JobTemplate)
-        original.unshift(display_link_if_authorized(_('Run'), hash_for_new_job_invocation_path(:template_id => template.id).merge(:authorizer => authorizer))) unless template.snippet
+        original.unshift(authorized_for(controller: :job_invocations, action: :create) ? link_to(_('Run'), hash_for_new_job_invocation_path(:template_id => template.id).merge(:authorizer => authorizer)) : nil) unless template.snippet
       end
```

Comment:
It might be just personal preference, but I'd prefer to avoid `condition ? something : nil` where reasonably possible. Could we move the `authorized_for` outside and either combine it with the unless at the end of the line or with the `template.is_a?` on the line above?

Reply:
I'm not sure how to do it while avoiding logic mistakes in this line. I can make a replacement function for the `display_link_if_authorized` that was there before and put the `authorized_for` condition there?

Reply:
Nevermind then, let's keep it as is and get this in. We shouldn't go ahead and invent our own flavor of helpers from foreman. Something tells me this is not the last time we're seeing something like this happening, so we may eventually want to fix this properly in foreman itself.